src/main.rs

```diff
@@ -148,8 +148,11 @@ fn main() -> Result<(), Error> {
     let kalman_handle = start_kalman(&mut communication_registry)?;
 
     let avg_handle = start_avg_filter(&mut communication_registry)?;
-    let (generated_data_handle, generator_handle) =
-        TrajectoryGenerator::run(1.0 / GENERATOR_FREQ, Arc::clone(&shutdown_trigger));
+    let (generated_data_handle, generator_handle) = TrajectoryGeneratorBuilder::new()
+        .with_perlin_mode()
+        .with_period(1.0 / GENERATOR_FREQ)
```

Comment:
Consider moving period calculation inside builder, so we need to specify only frequency here.
```suggestion
        .with_period(GENERATOR_FREQ)
```